### DISCUSSION TITLE
Succesive RigidBodyMotions

### DISCUSSION THREAD

**Discussion by Marcraven** [FABLE-3DXRD/xrd_simulator#8] [General]
I know the documentation says that motion takes place with respect to the origin of coordinates. But just to be 100% sure, if I rotate the sample 90 degrees over [0 0 1] and then rock it during diffraction 5 degrees on [0 1 0], this rotation axis is with respect the laboratory reference system right?

**Comment by AxelHenningsson** [comment]
Let's find out 😄 🤟 ❤️  !
----------------------------------


![movie](https://github.com/FABLE-3DXRD/xrd_simulator/assets/31615210/d467d8db-566c-4420-a258-492ec3ff16a0)

Generated in the following fashion:

```python

    m1 = RigidBodyMotion(rotation_axis=np.array([0, 0, 1]),
                        rotation_angle=np.radians(90),
                        translation=np.array([0, 0, 0]))

    m2 = RigidBodyMotion(rotation_axis=np.array([0, 1, 0]),
                        rotation_angle=np.radians(45),
                        translation=np.array([0, 0, 0]))

    for i in range(21):
        polycrystal.transform(m1, time=0.05)
        polycrystal.mesh_lab.save('sample_at_timestep_'+str(i).zfill(3)+'.xdmf')

    for j in range(11):
        polycrystal.transform(m2, time=0.1)
        polycrystal.mesh_lab.save('sample_at_timestep_'+str(j + i).zfill(3)+'.xdmf')
```
So - **Yes** ! The motion is always in relation to the laboratory coordinate system.

Cheers
Axel
